include/pisa/topk_queue.hpp

```diff
@@ -60,14 +60,21 @@ struct topk_queue {
 
     [[nodiscard]] std::vector<entry_type> const& topk() const noexcept { return m_q; }
 
-    void set_threshold(Threshold t) noexcept { m_threshold = t; }
+    void set_threshold(Threshold t) noexcept
```

Comment:
Name and docs suggestion:
```cpp
/// Override the threshold to be `threshold`.
///
/// This is meant to be used at the beginning of query processing if a threshold is available beforehand,
/// e.g., by means of estimation. Note that if `threshold` is higher than it would be at the end of query processing,
/// it could result in incomplete document list.
void force_threshold(Threshold threshold) noexcept
```